### DISCUSSION TITLE
Can you interact with the browser?

### DISCUSSION THREAD

**Discussion by yask123** [johnlindquist/kit#740] [Q&A] [answered]
I want to be able to open browser with a url, enter login credential and then scrape some data.

**Comment by johnlindquist** [comment, marked as answer]
There's a good thread here on fully interacting with Chrome:

https://github.com/johnlindquist/kit/discussions/629#discussioncomment-1725659

If you'd rather have it run in the background, I recommend using: https://www.npmjs.com/package/playwright

Let me know if you'd like an example.

**Comment by yask123** [comment]
Yea it'll be nice to have an example. Thanks!!

**Comment by johnlindquist** [comment]
@yask123 

Here's a full example. The `express` and `vue` stuff is setting up a mock server to scrape just as a demo.


[Open scrape-with-login in Script Kit](https://scriptkit.com/api/new?name=scrape-with-login&url=https://gist.githubusercontent.com/johnlindquist/99f95fb19bd86b4cb36f49b12e133782/raw/ee5126a5833da6c3d8f97c6fb814823c249f8522/scrape-with-login.ts")

```js
// Name: Scrape with Login

import "@johnlindquist/kit"

let port = 3579
let url = `http://localhost:${port}`

// Start - This is setting up a server to "login" and scrape
let express = await npm("express")
let app = express()

let html = `
<script src="https://unpkg.com/petite-vue" defer init></script>

<div v-scope="{ login: '' }">
  <input class="form" v-model="login"/>
  <p>{{login}}</p>
  <div class="find-me" v-if="login === 'john'">
    Hi John!
  </div>
</div>
`

app.get("/", (req, res) => {
    res.send(html)
})

let server = app.listen(port)

// if you want to check the page
// exec(`open '${url}'`)
// hide()

// End

let headless = false
let { chromium } = await npm("playwright")
let browser = await chromium.launch({ headless })
let context = await browser.newContext()
let page = await context.newPage()

await page.goto(url)
await page.fill('.form', 'john');

// Read docs on elementHandle: https://playwright.dev/docs/api/class-elementhandle
let handle = await page.$(".find-me")

let text = await handle.innerText()

// Shut down express
server.close()
// Shut down playwright
browser.close()

await editor(text)
```
